src/pages/ProfilePage/ProfilePage.jsx

```diff
@@ -1,10 +1,217 @@
-import React from 'react';
+import Post from 'components/common/Posts/Post';
+import {NormalButton} from 'components/UI/Buttons/buttons';
+import {IconCalendar, IconClose, IconShare} from 'components/UI/Icons/Icons';
+import { useAuth } from 'context/AuthContext';
+import {firestore} from 'firebase.config';
+import {doc, getDoc, updateDoc} from 'firebase/firestore';
+import React, {useEffect, useState, useReducer} from 'react';
+import {useParams} from 'react-router';
+import {getAllPost} from 'utils/postService';
 import "./ProfilePage.css";
 
+
+const profileReducerHandler = (state, action) => {
+    if (action.type === "name") {
+        return {
+            ...state,
+            name: action.name
+        }
+    }
+    if (action.type === "bio") {
+        return {
+            ...state,
+            bio: action.bio
+        }
+    }
+    return {
+        ...state
+    }
+}
 const ProfilePage = () => {
-  return (
-    <div>ProfilePage</div>
-  )
+  const { userID } = useParams();
+  const { userState, userDispatch } = useAuth();
+    const [userData, setUserData] = useState({});
+    const [yourPostArray, setYourPostArray] = useState([]);
+    const [editToggle, setEditToggle] = useState(false);
+    const [profilestate, profiledispatch] = useReducer(profileReducerHandler, {
+        name: "",
+        bio: ""
+    });
+    const [showCopied, setShowCopied] = useState(false);
+    console.log(userID);
+
+    useEffect(() => {
+        const userRef = doc(firestore, `users/${userID}`);
+        (async () => {
+            try {
+                const response = await getDoc(userRef);
+                console.log(response.data(), response.id, setUserData, userID);
+                console.log(response.data()[userID]);
+                setUserData(response.data()[userID]);
+                // setData(res1.data() ?? {});
+
+
+                // get All data
+                console.log(getAllPost());
+                let data = await getAllPost();
+                // if (data.isArray()) {
```

Comment:
If value is not re initialized, prefer using const

Reply:
thanks done